### PR TITLE
Make CC Cold Key a Credential

### LINKED ISSUE #3564: Constitutional Commitee Cold should be a Credential
Related to #3509 Cold key actually can be a script as well. We should reflect that in the registration and authorization certificates. As well as in scripts needed function

### PR DESCRIPTION
**Pull request by aniketd**
# Description

Resolves #3564 

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [x] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: eras/conway/impl/src/Cardano/Ledger/Conway/TxCert.hs [hunk 1]
```diff
@@ -520,7 +514,14 @@ getScriptWitnessConwayTxCert = \case
...
       ConwayDelegCert cred _ -> credScriptHash cred
       ConwayRegDelegCert cred _ _ -> credScriptHash cred
+  ConwayTxCertCommittee committeeCert -> committeeWitness committeeCert
   _ -> Nothing
+  where
+    committeeWitness :: ConwayCommitteeCert c -> Maybe (ScriptHash c)
+    committeeWitness = \case
+      ConwayAuthCommitteeHotKey _cold@(ScriptHashObj scriptHash) _hot -> Just scriptHash
+      ConwayResignCommitteeColdKey _cold@(ScriptHashObj scriptHash) -> Just scriptHash
+      _ -> Nothing
```
Comment (lehins) [inline eras/conway/impl/src/Cardano/Ledger/Conway/TxCert.hs]: Reduce the clutter and reuse specialized `credScriptHash` function
```suggestion
  ConwayTxCertCommittee committeeCert -> 
    case committeeCert of
      ConwayAuthCommitteeHotKey coldCred _hotCred -> credScriptHash coldCred
      ConwayResignCommitteeColdKey coldCred -> credScriptHash coldCred
  _ -> Nothing
```

File: eras/conway/impl/src/Cardano/Ledger/Conway/TxCert.hs [hunk 2]
```diff
@@ -532,4 +533,10 @@ getVKeyWitnessConwayTxCert = \case
...
       ConwayRegDelegCert cred _ _ -> credKeyHashWitness cred
   ConwayTxCertPool poolCert -> Just $ poolCertKeyHashWitness poolCert
-  ConwayTxCertCommittee committeeCert -> committeeKeyHashWitness committeeCert
+  ConwayTxCertCommittee committeeCert -> committeeWitness committeeCert
+  where
+    committeeWitness :: ConwayCommitteeCert c -> Maybe (KeyHash 'Witness c)
+    committeeWitness = \case
+      ConwayAuthCommitteeHotKey _cold@(KeyHashObj keyHash) _hot -> Just $ asWitness keyHash
+      ConwayResignCommitteeColdKey _cold@(KeyHashObj keyHash) -> Just $ asWitness keyHash
+      _ -> Nothing
```
Comment (lehins) [inline eras/conway/impl/src/Cardano/Ledger/Conway/TxCert.hs]: Same here:
```suggestion
  ConwayTxCertCommittee committeeCert -> 
    case committeeCert of
      ConwayAuthCommitteeHotKey coldCred _hotCred -> credKeyHashWitness coldCred
      ConwayResignCommitteeColdKey coldCred -> credKeyHashWitness coldCred
      _ -> Nothing
```

**Review by lehins — CHANGES_REQUESTED** [review]
Looks good. Couple of suggestions and you are still missing changes to the CDDL spec

**Review by lehins — APPROVED** [review]
:+1:

File: eras/conway/impl/CHANGELOG.md [hunk 3]
```diff
@@ -3,13 +3,13 @@
 ## 1.7.0.0
 
 * Add an optional Anchor to the Conway DRep registration certificate #3576
+* Change `ConwayCommitteeCert` to allow:
+  * committee cold keys to be script-hashes #3581
+  * committee hot keys to be script-hashes #3552
```
Comment (aniketd) [inline eras/conway/impl/CHANGELOG.md:8]: @lehins noted! :memo: :raised_hands: Thank you!

**Pull request merged**